libarchive/archive_write_set_format_cpio_newc.c

```diff
@@ -190,7 +190,7 @@ archive_write_newc_header(struct archive_write *a, struct archive_entry *entry)
 	const char *path;
 	size_t len;
 
-	if (archive_entry_filetype(entry) == 0) {
+	if (archive_entry_filetype(entry) == 0 && archive_entry_hardlink_is_set(entry) == 0) {
```

Comment:
This is the right idea, I think.  We should keep this check, since people trying to create archives programmatically can easily overlook important bits.  You might consider using `archive_entry_hardlink(entry) == NULL` as the test here (as is used a few lines below) instead of adding a new API.

Reply:
Oh, how did I miss that function?  Thank you!